src/npc.cpp

```diff
@@ -256,7 +256,7 @@ void Npc::onCreatureAppear(Creature* creature, bool isLogin)
 		if (npcEventHandler) {
 			npcEventHandler->onCreatureAppear(creature);
 		}
-	} else if (Player* player = creature->getPlayer()) {
+	} else if (std::shared_ptr<Player> player = creature->getPlayer()) {
```

Comment:
Could this be `const auto`
```suggestion
	} else if (const auto player = creature->getPlayer()) {
```